lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1568,48 +1567,30 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
       // Assuming that the bitwidth and FloatTy->getFPKind() are identical.
       SizeInBits = FloatTy->getBitWidth();
       Encoding = llvm::dwarf::DW_ATE_float;
-      if (auto DbgTyNumExtraInhabitants = DbgTy.getNumExtraInhabitants())
-        NumExtraInhabitants = *DbgTyNumExtraInhabitants;
       break;
     }
 
-    case TypeKind::BuiltinNativeObject: {
-      unsigned PtrSize = CI.getTargetInfo().getPointerWidth(clang::LangAS::Default);
-      auto PTy = DBuilder.createPointerType(nullptr, PtrSize, 0,
-                                            /* DWARFAddressSpace */ llvm::None,
-                                            MangledName);
-      return DBuilder.createObjectPointerType(PTy);
-    }
-
-    case TypeKind::BuiltinBridgeObject: {
-      unsigned PtrSize = CI.getTargetInfo().getPointerWidth(clang::LangAS::Default);
-      auto PTy = DBuilder.createPointerType(nullptr, PtrSize, 0,
-                                            /* DWARFAddressSpace */ llvm::None,
-                                            MangledName);
-      return DBuilder.createObjectPointerType(PTy);
-    }
-
-    case TypeKind::BuiltinRawPointer: {
-      unsigned PtrSize = CI.getTargetInfo().getPointerWidth(clang::LangAS::Default);
-      return DBuilder.createPointerType(nullptr, PtrSize, 0,
-                                        /* DWARFAddressSpace */ llvm::None,
-                                        MangledName);
-    }
+    case TypeKind::BuiltinNativeObject: 
+    case TypeKind::BuiltinBridgeObject:
+    case TypeKind::BuiltinRawPointer:
+    case TypeKind::BuiltinRawUnsafeContinuation:
+    case TypeKind::BuiltinJob: {
+      unsigned PtrSize =
+          CI.getTargetInfo().getPointerWidth(clang::LangAS::Default);
+      if (Opts.DebugInfoLevel > IRGenDebugInfoLevel::ASTTypes) {
+        Flags |= llvm::DINode::FlagArtificial;
+        llvm::DICompositeType *PTy = DBuilder.createStructType(
+            Scope, MangledName, File, 0, PtrSize, 0, Flags, nullptr, nullptr,
+            llvm::dwarf::DW_LANG_Swift, nullptr, {}, NumExtraInhabitants);
+        return PTy;
 
-    case TypeKind::BuiltinRawUnsafeContinuation: {
-      unsigned PtrSize = CI.getTargetInfo().getPointerWidth(clang::LangAS::Default);
-      return DBuilder.createPointerType(nullptr, PtrSize, 0,
-                                        /* DWARFAddressSpace */ llvm::None,
-                                        MangledName);
-    }
+      }
+      llvm::DIDerivedType *PTy = DBuilder.createPointerType(
+          nullptr, PtrSize, 0,
+          /* DWARFAddressSpace */ llvm::None, MangledName);
 
-    case TypeKind::BuiltinJob: {
-      unsigned PtrSize = CI.getTargetInfo().getPointerWidth(clang::LangAS::Default);
-      return DBuilder.createPointerType(nullptr, PtrSize, 0,
-                                        /* DWARFAddressSpace */ llvm::None,
-                                        MangledName);
+      return DBuilder.createObjectPointerType(PTy);
```

Comment:
Does this intentionally change the behavior for BuiltinRawPointer?

Reply:
Yes, I changed it because all of them seem artificial types to me (builtins not visible in user source code). What do you think?